tests/test_integrators.py

```diff
@@ -151,30 +151,36 @@ def test_npt_langevin(ar_double_sim_state: ts.SimState, lj_model: LennardJonesMo
 def test_npt_langevin_multi_kt(
     ar_double_sim_state: ts.SimState, lj_model: LennardJonesModel
 ):
-    dtype = torch.float64
     n_steps = 200
-    dt = torch.tensor(0.001, dtype=dtype)
-    kT = torch.tensor([300, 10_000], dtype=dtype) * MetalUnits.temperature
-    external_pressure = torch.tensor(0, dtype=dtype) * MetalUnits.pressure
-
-    # Initialize integrator
-    init_fn, update_fn = npt_langevin(
-        model=lj_model,
-        dt=dt,
-        kT=kT,
-        external_pressure=external_pressure,
-        alpha=40 * dt,
+    dt = torch.tensor(0.001, dtype=DTYPE)
+    kT = torch.tensor([300, 10_000], dtype=DTYPE) * MetalUnits.temperature
+    external_pressure = torch.tensor(0, dtype=DTYPE) * MetalUnits.pressure
+    alpha = 40 * dt
+    cell_alpha = alpha
+    b_tau = 1 / (1000 * dt)
+
+    # Initialize integrator using new direct API
+    state = ts.npt_langevin_init(
+        model=lj_model, state=ar_double_sim_state, dt=dt, kT=kT, alpha=alpha, seed=42
     )
 
     # Run dynamics for several steps
-    state = init_fn(state=ar_double_sim_state, seed=42)
     energies = []
     temperatures = []
     for _step in range(n_steps):
-        state = update_fn(state=state)
+        state = ts.npt_langevin_update(
```

Comment:
feels like the refactor likely introduces a bug here in that we should be passing a generator to this update function?

Reply:
I would think (not sure/haven't/won't check) before the init function that returned the update function would have made a generator in the factory that was used for the PRNG sequence inside the update. Here this is unseeded which is probably fine for now but would be better to fix alongside something like: https://github.com/TorchSim/torch-sim/issues/51

Reply:
That would have been the right way to do it but not what we did. Fixing the seeds doesn't really work on most integrators beyond init. Agree it can be addressed later.